engine/src/main/java/org/terasology/engine/rendering/assets/mesh/resource/VertexAttribute.java

```diff
@@ -0,0 +1,64 @@
+// Copyright 2021 The Terasology Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+package org.terasology.engine.rendering.assets.mesh.resource;
+
+import org.joml.Vector2f;
+import org.joml.Vector3f;
+import org.joml.Vector4f;
+import org.lwjgl.opengl.GL30;
+import org.terasology.nui.Color;
+
+import java.nio.ByteBuffer;
+
+/**
+ * attribute maps a target object or a primitive data to a {@link VertexResource}
+ *
+ * @param <TARGET> the target object
+ */
+public class VertexAttribute<TARGET> {
```

Comment:
Hm, this reads funny. I guess I'm just not used to having full words :scream: as type parameters... I guess you'd typically just use `T` in Java. What is the reason you chose `TARGET`?